### PR TITLE
gcj: fix build on Darwin

### PR DESCRIPTION
**Pull request by alyssais**
###### Motivation for this change

Partially addresses https://github.com/NixOS/nixpkgs/issues/29715.
Part of #54573.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by matthewbauer** [comment]
Awesome work!

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -7012,7 +7012,9 @@ in
     profiledCompiler = false;
     inherit zip unzip zlib boehmgc gettext pkgconfig perl;
     inherit (gnome2) libart_lgpl;
-  });
+  }).overrideAttrs (oldAttrs: {
+    strictDeps = stdenv.isDarwin;
```
Comment (matthewbauer) [inline pkgs/top-level/all-packages.nix]: IIRC gcc has strictDeps=true alread?
/cc @Ericson2314

Comment (alyssais) [reply pkgs/top-level/all-packages.nix]: If it does, it didn't in November when I first authored this commit. Although this should definitely be `oldAttrs.strictDeps or stdenv.isDarwin`.

Comment (alyssais) [reply pkgs/top-level/all-packages.nix]: You were right! We didn't need this, so I've dropped this change.

**Review by LnL7 — APPROVED** [review]

**Pull request merged**